### PR TITLE
Alternative checkout example

### PR DESCRIPTION
**Pull request by Kagaminara**
This PR adds an "alternative checkout flow" example, which merges the two first steps; Address and Shipping method selection.

Note that it is a "naive" implementation, and although working, it only serves as an example as to how to modify the checkout flow.

The enhancers have been overriden to move the mutation part to the parent `AddressAndShippingMethod` component.

### CODE REVIEW

File: alternative-checkout/web/theme/modules/Checkout/Address/ExistingAddress/ExistingAddress.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      setShipping(data);
+      setError(null);
+    } else if (type === "billing") {
+      setBilling(data);
+    } else if (type === "shipping") {
+      setShipping(data);
+    } else {
+      setError(error);
+    }
+    props.onAddressChange();
```
Comment (Kagaminara) [inline alternative-checkout/web/theme/modules/Checkout/Address/ExistingAddress/ExistingAddress.js:70]: The main change in this component is this line. It is used to "invalidate" the shipping method, in order to prevent customer to go to the next step while changing addresses.

File: alternative-checkout/web/theme/modules/Checkout/ShippingMethod/ChooseShippingMethod.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        <span>
+          <FormattedMessage
+            id="modules.Checkout.ShippingMethod.ChooseShippingMethod.noShippingAddress"
+            defaultMessage="Please select a shipping address before choosing a shipping method."
+          />
+        </span>
+      </InfoAlert>
+    </TitledCard>
+  );
+};
```
Comment (Kagaminara) [inline alternative-checkout/web/theme/modules/Checkout/ShippingMethod/ChooseShippingMethod.js:54]: One of the addition in this "step". Since Shipping methods part is displayed at all time, we now display an information (instead of an error), if the shipping address is missing.

File: alternative-checkout/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     { name: "FrontCommerce", path: "server/modules/front-commerce" },
+```
+```diff
+--- a/.front-commerce.js
++++ b/.front-commerce.js
+@@ -18,5 +19,6 @@ module.exports = {
+     { name: "FrontCommerce", path: "front-commerce/src/web" },
+     { name: "ThemeChocolatine", path: "front-commerce/theme-chocolatine/web" },
+     { name: "Skeleton", path: "./src/web" },
++    { name: "AlternativeCheckout", path: "./modules/alternative-checkout/web" },
```
Comment (real34) [inline alternative-checkout/README.md]: Shouldn't it be `./examples`?
```suggestion
+    { name: "AlternativeCheckout", path: "./examples/alternative-checkout/web" },
```

File: alternative-checkout/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Alternative checkout
+
+This example shows a way to alter the checkout flow by "merging" the address and shipping method steps from the checkout, while reusing most of Front-Commerce components.
+
+This example shows different practical use of [Configuration Providers](https://developers.front-commerce.com/docs/2.x/advanced/server/configurations#what-is-a-configuration-provider) in Front-Commerce.
```
Comment (real34) [inline alternative-checkout/README.md]: **issue:** seems unrelated (copy/paste?)

**suggestion:** remove
```suggestion
```

**Review by real34 — COMMENTED** [review]
Quick feedbacks from overview

File: alternative-checkout/README.md (hunk 4 above)
Comment (Kagaminara) [reply alternative-checkout/README.md]: Indeed ! I pasted it to have an example under my eye and forgot to remove it 🤦
